pkg/resolvers/resolvers.go

```diff
@@ -0,0 +1,58 @@
+// Package resolvers resolves what rules different users and roleTemplates our bound to
+package resolvers
+
+import (
+	"fmt"
+
+	rbacv1 "k8s.io/api/rbac/v1"
+)
+
+// ErrUnimplemented is an error returned when a function is not implemented.
+var ErrUnimplemented = fmt.Errorf("not implemented")
+
+// ruleAccumulator based off kubernetes struct
+// https://github.com/kubernetes/kubernetes/blob/d5fdf3135e7c99e5f81e67986ae930f6a2ffb047/pkg/registry/rbac/validation/rule.go#L124#L137
+type ruleAccumulator struct {
+	rules  []rbacv1.PolicyRule
```

Comment:
I think in this case it may be a good idea to use a slice of pointers over values. There will be tons of rules, many of which redundant I imagine, in a scaled setup. Adding another object to memory every time we append won't be necessary and we aren't writing to the rules or doing anything dangerous like that.

Reply:
We would not save any space using pointers here since this list is ultimately returned by the `RulesFor` method which must adhere to the [validation.AuthroizationRuleResolver](https://github.com/kubernetes/kubernetes/blob/500f5aba80d71253cc01ac6a8622b8377f4a7ef9/pkg/registry/rbac/validation/rule.go#L36) interface.  This means we would have to inevitably allocate memory for each item in the list eventually.

An alternative to combat memory usage we have the option of using a map internal for the accumulator. This map would use `rbacv1.PolicyRule.String()` as the key so that we do not store duplicate rules and errors. This approach may save us some memory, but when benchmarked it caused significant speed decrease
```
goos: darwin
goarch: arm64
pkg: github.com/rancher/webhook/pkg/resolvers
BenchmarkGotRules-10                        3807            274032 ns/op
BenchmarkGotRulesWithMapCheck-10             852           1195656 ns/op
BenchmarkGotRulesWithMapOverride-10          993           1255381 ns/op
```
 **note this BenchMark resolved rules for users with 10 roleTemplates that each contained 100 rules.**

Since this list is evaluated then discarded the total memory usage should not be a large concern. What we really care about is the added time needed to allocate extra memory.